### ISSUE TITLE
Replace by fee calculates new fee incorrectly

### ISSUE THREAD

**Issue by friedger** [blockstack/stacks-blockchain#2481]
**Describe the bug**
Will trying to replace a tx, it was rejected until I increased the fees drastically, like 10x

From conversation with @kantai :
```
message
    TX conflicts with sponsor/origin nonce in same fork with >= fee
new_fee
    37,800,000
new_txid
    910027a75cc93e954f4188a095bd93655c142acaecadc88180325c7ec957daa6
old_fee
    222,378,000
old_txid
    db6188708ceed6a957bf0a5ff9c4ffff23b4e0ec50ac510f8cec3cd4d0c27f14
```

**Comment by jcnelson** [comment]
This sounds like there's some left-over code in the mempool that still treats the transaction fee as a fee _rate_, and incorrectly multiplies it by the transaction's byte length.

**Comment by zone117x** [comment]
@jcnelson `fee_rate` is still used in (at least) variable names within the API. Should they all be `fee` instead?

**Comment by jcnelson** [comment]
Sure, provided it doesn't lead to breakage.  It's called `fee` internally in the Stacks blockchain already.

**Comment by pavitthrap** [comment]
Fixed in #2508
